### ISSUE TITLE
ts server crashing

### ISSUE THREAD

**Issue by fafo** [microsoft/vscode#170526]
Type: <b>Bug</b>

TS server continously crashes while loading teh project and no autocompletion is avalialble

VS Code version: Code 1.74.2 (e8a3071ea4344d9d48ef8a4df2c097372b0c5161, 2022-12-20T10:27:19.253Z)
OS version: Linux x64 5.15.0-56-generic
Modes:
Sandboxed: Yes

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-10750H CPU @ 2.60GHz (12 x 2600)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: disabled_software<br>video_encode: disabled_software<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: disabled_off|
|Load (avg)|6, 6, 6|
|Memory (System)|39.00GB (26.91GB free)|
|Process Argv|--unity-launch --crash-reporter-id 345d115c-2013-43a7-89fa-5068bd7ae219|
|Screen Reader|no|
|VM|0%|
|DESKTOP_SESSION|gnome|
|XDG_CURRENT_DESKTOP|GNOME|
|XDG_SESSION_DESKTOP|gnome|
|XDG_SESSION_TYPE|x11|
</details><details><summary>Extensions (40)</summary>

Extension|Author (truncated)|Version
---|---|---
vscode-expo|byC|0.9.2
githistory|don|0.6.19
gitlens|eam|13.2.0
code-runner|for|0.11.8
copilot|Git|1.65.7705
cmake-language-support-vscode|jos|0.0.8
Kotlin|mat|1.7.1
git-graph|mhu|1.30.0
mongodb-vscode|mon|0.9.5
vscode-docker|ms-|1.23.3
vscode-dotnet-runtime|ms-|1.6.0
vscode-postgresql|ms-|0.3.0
isort|ms-|2022.8.0
python|ms-|2022.20.1
vscode-pylance|ms-|2022.12.20
jupyter|ms-|2022.11.1003412109
jupyter-keymap|ms-|1.0.0
jupyter-renderers|ms-|1.0.12
vscode-jupyter-cell-tags|ms-|0.1.6
vscode-jupyter-slideshow|ms-|0.1.5
remote-containers|ms-|0.266.1
cmake-tools|ms-|1.12.27
cpptools|ms-|1.13.8
vscode-typescript-next|ms-|5.0.202301020
sqltools|mtx|0.26.0
sqltools-driver-pg|mtx|0.4.0
env-cmd-file-syntax|Nix|0.3.0
vetur|oct|0.36.1
java|red|1.13.0
rust-analyzer|rus|0.3.1348
cmake|twx|0.0.17
vscode-lldb|vad|1.8.1
vscode-java-debug|vsc|0.47.0
vscode-java-dependency|vsc|0.21.1
vscode-java-pack|vsc|0.25.7
vscode-java-test|vsc|0.37.1
vscode-maven|vsc|0.40.2
vim|vsc|1.24.3
markdown-all-in-one|yzh|3.5.0
json|Zai|2.0.2

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
vslsvsres303:30308271
pythonvspyl392:30443607
vserr242:30382549
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vsdfh931cf:30280410
vshan820:30294714
vstes263:30335439
pythondataviewer:30285071
vscod805:30301674
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
cmake_vspar411:30581797
vsaa593cf:30376535
pythonvs932:30410667
cppdebug:30492333
vsclangdf:30486550
c4g48928:30535728
dsvsc012cf:30540253
azure-dev_surveyone:30548225
pyindex848cf:30577861
nodejswelcome1cf:30587006
2e4cg342:30602488
iaj6b796:30613358
f6dab269:30613381
fim-prod:30623723
vscsb:30628654

```

</details>

**Comment by mjbvz** [comment]
Does this reproduce in the [latest VS Code insiders build](https://code.visualstudio.com/insiders/) with all [extensions disabled](https://github.com/microsoft/vscode/wiki/Performance-Issues#run-with-extensions-disabled)?

**Comment by fafo** [comment]
Found the problem: in the project assets there was a video with extension .ts and this made the typescript server crash. 
Perhaps it should be better to look not only at the extension of the files, and to try guessing the mime type as the file command works. 
Now I've solved moving the file outside the project dir. 
I close the issue, but better mime type identification would be an improvement nice to have.
